marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -243,34 +259,25 @@ instance IsValid Context where
 
 -- | An arbitrary positive integer, mostly small.
 arbitraryPositiveInteger :: Gen Integer
-arbitraryPositiveInteger =
-  frequency
-    [
-      (60, arbitrary `suchThat` (> 0))
-    , (30, arbitraryFibonacci fibonaccis)
-    ]
+arbitraryPositiveInteger = arbitraryNonnegativeInteger <&> \case
+  0 -> 1
+  n -> n
 
 
 -- | An arbitrary non-negative integer, mostly small.
 arbitraryNonnegativeInteger :: Gen Integer
-arbitraryNonnegativeInteger =
-  frequency
-    [
-      (60, arbitrary `suchThat` (>= 0))
-    , (30, arbitraryFibonacci fibonaccis)
-    ]
+arbitraryNonnegativeInteger = frequency
+  [ (1, abs <$> arbitrary)
+  , (6, floor . sqrt @Double . abs <$> arbitrary)
+  ]
```

Comment:
Rationale:
- Generate only valid values instead of using `suchThat` to improve performance.
- To satisfy "mostly small", generate values on an inverse-quadratic curve instead of picking only Fibbonacci values (results in better coverage).